Backport -march=native support for Ivy Bridge (core-avx-i).

http://gcc.gnu.org/PR57927


commit e5a02c15d9f687ee7fbfb37c10ef645db3cbc09d
Author: uros <uros@138bc75d-0d04-0410-961f-82ee72b054a4>
Date:   Wed Aug 28 08:36:37 2013 +0000

    	Backport from mainline
    	2013-08-27  Christian Widmer  <shadow@umbrox.de>
    
    	PR target/57927
    	* config/i386/driver-i386.c (host_detect_local_cpu): Add detection
    	of Ivy Bridge processors.


    git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_7-branch@202047 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog                 |  9 +++++++++
 gcc/config/i386/driver-i386.c | 21 +++++++++++----------
 2 files changed, 20 insertions(+), 10 deletions(-)


--- a/gcc/config/i386/driver-i386.c
+++ b/gcc/config/i386/driver-i386.c
@@ -630,13 +630,18 @@ const char *host_detect_local_cpu (int argc, const char **argv)
 	  /* Atom.  */
 	  cpu = "atom";
 	  break;
+	case 0x0f:
+	  /* Merom.  */
+	case 0x17:
+	case 0x1d:
+	  /* Penryn.  */
+	  cpu = "core2";
+	  break;
 	case 0x1a:
 	case 0x1e:
 	case 0x1f:
 	case 0x2e:
 	  /* Nehalem.  */
-	  cpu = "corei7";
-	  break;
 	case 0x25:
 	case 0x2c:
 	case 0x2f:
@@ -648,14 +653,10 @@ const char *host_detect_local_cpu (int argc, const char **argv)
 	  /* Sandy Bridge.  */
 	  cpu = "corei7-avx";
 	  break;
-	case 0x17:
-	case 0x1d:
-	  /* Penryn.  */
-	  cpu = "core2";
-	  break;
-	case 0x0f:
-	  /* Merom.  */
-	  cpu = "core2";
+	case 0x3a:
+	case 0x3e:
+	  /* Ivy Bridge.  */
+	  cpu = "core-avx-i";
 	  break;
 	default:
 	  if (arch)
